proto/internal/temporal/server/api/worker_deployment/v1/message.proto

```diff
@@ -0,0 +1,183 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package temporal.server.api.worker_deployment.v1;
+
+option go_package = "go.temporal.io/server/api/worker_deployment/v1;worker_deployment";
+
+import "temporal/api/enums/v1/task_queue.proto";
+import "google/protobuf/timestamp.proto";
+import "temporal/api/deployment/v1/message.proto";
+import "temporal/api/common/v1/message.proto";
+
+// Data for each worker deployment version + task queue pair.
+// This is stored in each worker deployment version (for each task queue),
+// and synced to task queue user data (for each worker deployment version).
+message TaskQueueData {
+    google.protobuf.Timestamp first_poller_time = 1;
+    google.protobuf.Timestamp last_became_current_time = 2;
+}
+
+// Local state for Worker Deployment Version
+message VersionLocalState {
+    temporal.api.deployment.v1.Deployment deployment = 1;
+    map<string, TaskQueueFamilyData> task_queue_families = 2;
+
+    bool is_current = 3;
+    google.protobuf.Timestamp last_became_current_time = 6;
+
+    google.protobuf.Timestamp create_time = 4;
+    map<string, temporal.api.common.v1.Payload> metadata = 5;
+
+    bool started_series_workflow = 7;
+
+    message TaskQueueFamilyData {
+        // Key: enums.TaskQueueType, Value: TaskQueueData
+        // All fields in TaskQueueData are valid except last_became_current_time, the source of
+        // truth for that is the value at the root of DeploymentLocalState.
+        map<int32, TaskQueueData> task_queues = 1;
+    }
+}
+
+// used as Worker Deployment Version workflow input:
+message VersionWorkflowArgs {
+    string namespace_name = 1;
+    string namespace_id = 2;
+    VersionLocalState version_state = 3;
+}
+
+// used as Worker Deployment workflow input:
+message WorkflowArgs {
```

Comment:
I know the package name is worker_deployment, but WorkflowArg still feels too generic. I'd prefer DeploymentWorkflowArgs to clarify we're talking about a particular wf type.

Reply:
I went with `WorkerDeploymentWorkflowArgs` / `WorkerDeploymentWorkflow` and `WorkerDeploymentVersionWorkflowArgs` / `WorkerDeploymentVersionWorkflow`. I agree that for workflow type it makes sense to be extra specific

Reply:
I addressed and resolved all your comments to rename the fields within the proto messages and also remove fields that we're not implementing right away :)

They were all unchanged because Shivam and I were planning to merge this PR and then change them alongside Shivam's API PR. I think the fields left are ones we will use, but not 100% sure.